### PR TITLE
upload files

### PR DESCRIPTION
**Pull request by hayux**

### CODE REVIEW

File: codeUassign1q1 [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+"""
+Created on Mon May 15 21:24:05 2017
+
+@author: hexu
+"""
+
+def strPermuTest(str1, str2):
```
Comment (ariava) [inline codeUassign1q1]: This is missing tests. Could you please add some?

File: codeUassign1q2 [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+# a try        
+myList = SinglyLinkedList()
+myList.add(12)
+myList.add(14)
+myList.add(53)
+myList.add(31)
+
+
+myNode = myList.findNode(8)
+print(myNode.value)
```
Comment (ariava) [inline codeUassign1q2]: You can use the "assert" statement to perform automated checks (see e.g. https://www.tutorialspoint.com/python/assertions_in_python.htm). The assert statement breaks the code execution if any of the checks fail, throwing an AssertionError.

File: codeUassign1q1.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+"""
+Created on Mon May 15 21:24:05 2017
+
+@author: hexu
+"""
+
+def strPermuTest(str1, str2):
+    return sorted(str1.lower()) == sorted(str2.lower())
```
Comment (ariava) [inline codeUassign1q1.py]: This is very pythonic by the way :) Nice!

File: codeUassign1q1.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+# test cases
+print(strPermuTest('lisTnendg','NenGLisd'))
+print(strPermuTest('bala lalab','labla xddd'))
```
Comment (ariava) [inline codeUassign1q1.py]: Cool! Maybe next time try to use the assert statement (https://www.tutorialspoint.com/python/assertions_in_python.htm).

**Comment by hayux** [comment]
Thank you :) I'm reading the pages you sent me and learning how to do it
now :)

On Tue, May 23, 2017 at 10:35 PM, Arianna Avanzini <notifications@github.com

File: codeUassign1q2 (hunk 2 above)
Comment (hayux) [inline codeUassign1q2]: I just tried to throw an exception on checking the inputs :) hope it works

File: codeUassign1q2.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+# a try        
+myList = SinglyLinkedList()
+myList.add(12)
+myList.add(14)
+myList.add(53)
+myList.add(31)
+
+
+myNode = myList.findNode(8)
+print(myNode.value)
```
Comment (ariava) [inline codeUassign1q2.py]: Good job for using assert!

Can I suggest you add a couple automated tests, e.g.:

assert myList.findNode(1).value == 12
assert myList.findNode(4).value == 31
assert myList.findNode(0).value == 31

and/or others you think are meaningful? The idea is to test the corner cases of your code, pushing it to the limit. This provides a good coverage and good documentation of your code (documentation which you will be forced to update while you develop your code! :))

File: codeUassign1q1.py (hunk 4 above)
Comment (ariava) [inline codeUassign1q1.py]: Example of using assert here:

assert not strPermuTest('lisTnendg', 'NenGLisd')
assert not strPermuTest('bala lalab', 'labla xddd')

and maybe also add a case where the two are actually permutations.
